core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -74,6 +76,21 @@ object Battle {
             attacker.unit.action = UnitActionType.SetUp.value
             attacker.unit.useMovementPoints(1f)
         }
+
+        if (tryHealPillage) {
+            // Now lets retroactively see if we can pillage any improvement on the path improvement to heal
+            // While still being able to attack
+            val pillageIterator = tilesMovedThrough.toMutableList().listIterator()
```

Comment:
There is *really* no need for an iterator here, works fine for 'just a list' and less confusing that way (list.first(), list.remove())